index.js

```diff
@@ -29,67 +28,64 @@ DatabaseUpdates.prototype = Object.create(EventEmitter.prototype)
 
 DatabaseUpdates.prototype.getFiles = function getFiles() {
   try {
-    this.updateFiles = fs.readdirSync(this.updatePath)
+    this.updateFiles = fs
```

Comment:
curious why semver major version dictates the list of pending "data migrations" - versus a `applied` boolean (or `dateApplied` timestamp. or, the inverse, an enum: `"pending" | "success" | "error"`.

that's the pattern I'm familiar with, but the current behaviour does make logical; breaking changes necessitate a major semver upgrade, which necessitates running a new migration file.

knowing which ones have been applied, and which ones have yet to be applied, feels safer from user error. I have to remember to bump the package every time a migration is added. accidents happen. without the guardrails, a migration could mistakingly not be run, whilst code is deployed expecting the schema/data update to have been applied.

to add another layer of paranoid caution: data migrations and human mistakes can go hand in hand. so, beyond storing the `dateApplied` timestamp, storing a hash of the file content (like [`migrate-mongo` does](https://github.com/seppevs/migrate-mongo/blob/master/lib/actions/up.js#L38-L44)) - not just the filename - could prove useful to those botched the migration, bumped the package version, and shipped it to production.

take all this with a grain of salt. I'm relaying this only from past incidents I've encountered. with carefulness, none of this should be necessary.

Reply:
The file hash idea is a good one! If we wanna use this, i'm happy to build it out, or PRs welcome.